src/main/java/org/broadinstitute/hellbender/tools/exome/ReadCountCollectionUtils.java

```diff
@@ -312,12 +408,235 @@ public static String getSampleNameFromReadCounts(final ReadCountCollection count
                 writer.writeRecord(new ReadCountRecord.SingleSampleRecord(new Target(interval), coverage));
             }
         } catch (final IOException e) {
-            throw new UserException.CouldNotCreateOutputFile(outFile, e);
+            throw new UserException.CouldNotCreateOutputFile(outName, e);
         }
 
+    }
 
+    /**
+     * Impute zero counts to the median of non-zero values in the enclosing target row.
+     *
+     * <p>The imputation is done in-place, thus the input matrix is well be modified as a result of this call.</p>
+     *
+     * @param readCounts the input and output read-count matrix.
+     */
+    public static void imputeZeroCountsAsTargetMedians(final ReadCountCollection readCounts, final Logger logger) {
+
+        final RealMatrix counts = readCounts.counts();
+        final int targetCount = counts.getRowDimension();
+
+        final Median medianCalculator = new Median();
+        int totalCounts = counts.getColumnDimension() * counts.getRowDimension();
+
+        // Get the number of zeroes contained in the counts.
+        final long totalZeroCounts = IntStream.range(0, targetCount)
+                .mapToLong(t -> DoubleStream.of(counts.getRow(t))
+                        .filter(c -> c == 0.0).count()).sum();
+
+        // Get the median of each row, not including any entries that are zero.
+        final double[] medians = IntStream.range(0, targetCount)
+                .mapToDouble(t -> medianCalculator.evaluate(
+                        DoubleStream.of(counts.getRow(t))
+                                .filter(c -> c != 0.0)
+                                .toArray()
+                )).toArray();
+
+        // Change any zeros in the counts to the median for the row.
+        counts.walkInOptimizedOrder(new DefaultRealMatrixChangingVisitor() {
+            @Override
+            public double visit(final int row, final int column, final double value) {
+                return value != 0 ? value : medians[row];
+            }
+        });
```

Comment:
For two reasons: First, `RealMatrix` is not an `Collection` like `List`, `Set`, etc., and it does not support `stream()`. It doesn't even make sense for them to be streamable: `RealMatrix` is an abstract class and different extensions store data in different ways (dense matrices, sparse matrices, etc) and it is not possible to stream their data in a meaningful way. Second, `walkInOptimizerOrder` takes an instance of `RealMatrixChangingVisitor` which is **not** a functional interface (functional interfaces many only have one abstract method; `RealMatrixChangingVisitor` has three). A lambda expression in Java 8 is basically a syntactic sugar for instantiating an anonymous functional interface.